dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/GroupCommParameterServerTask.java

```diff
@@ -79,17 +81,23 @@ private GroupCommParameterServerTask(
     this.validationStatsPairReduceReceiver =
         commGroup.getReduceReceiver(NeuralNetworkGroupCommDriver.ValidationStatsPairReduce.class);
 
+    this.matrixFactory = matrixFactory;
     this.layerParameters = new LayerParameter[serializedLayerConfigurationSet.size()];
     this.deltaLayerParameters = new LayerParameter[serializedLayerConfigurationSet.size()];
     this.stepsize = stepsize;
     this.maxIterations = maxIterations;
 
+    final Configuration matrixFactoryConf = Tang.Factory.getTang().newConfigurationBuilder()
+        .bindImplementation(MatrixFactory.class, matrixFactory.getClass())
+        .build();
+
     for (final String serializedInitializerConfiguration : serializedLayerConfigurationSet) {
       try {
         final Configuration initializerConfiguration =
             configurationSerializer.fromString(serializedInitializerConfiguration);
         final LayerParameterInitializer layerParameterInitializer =
-            Tang.Factory.getTang().newInjector(initializerConfiguration).getInstance(LayerParameterInitializer.class);
+            Tang.Factory.getTang().newInjector(initializerConfiguration, matrixFactoryConf)
```

Comment:
Can we receive an Injector at the constructor and call `injector.forkInjector()` here? That way, we don't have to create `matrixFactoryConf`.